### PR TITLE
raft: improve logging around tick miss

### PR DESCRIPTION
**Pull request by gyuho**
To help debug https://github.com/etcd-io/etcd/issues/9939.

@tbg @xiang90 @jingyih @manishrjain

### CODE REVIEW

File: raft/node.go [hunk 1]
```diff
@@ -261,6 +265,9 @@ type node struct {
 	stop       chan struct{}
 	status     chan chan Status
 
+	id       func() uint64
```
Comment (tbg) [inline raft/node.go]: Let's just put the`*RawNode` here. Then we can also remove `logger` and you don't need to pass the RawNode to `run` any more. The current setup is quite confusing.

**Comment by tbg** [comment]
PS it's an antipattern to actually do heavy lifting in `.Snapshot` since it blocks the event loop. The way we do it in CockroachDB is that `.Snapshot()` basically just returns an empty slice. When the corresponding `MsgSnap` is returned from the ready, we populate it with the snapshot and send it out. Actually we also apply rate limiting, streaming, etc because snapshots can be large.

**Comment by gyuho** [comment]
@tbg PTAL. Thanks!

**Comment by tbg** [comment]
Looks good, thanks @gyuho

**Comment by manishrjain** [comment]
> I think the best way for dgraph to get to the bottom of their issues is to instrument their Storage implementation and log a warning with the actual duration whenever anything takes upwards of a few ticks, perhaps ~400ms in their setting. I expect that they'll see .Snapshot take multiple seconds, but it could also be .Entries(); who knows.

We started with switching the ticks to be 100ms instead of 20ms, to ease off the pressure a bit. We have cache around Snapshot within the Raft storage implementation, so that shouldn't be slow. But, can add one around Entries to see how that goes, if we still see the missed to fire after the tick duration change.

**Pull request merged**